### PR TITLE
RavenDB-12889 Provide the user to find an ongoing task by name

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 1]
```diff
@@ -851,7 +858,10 @@ public Task GetOngoingTaskInfo()
                     {
                         case OngoingTaskType.Replication:
 
-                            var watcher = record.ExternalReplications.Find(x => x.TaskId == key);
+                            var watcher = name != null ?
```
Comment (maximburyak) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:861]: I noticed that we have a "taskName" query param for subscriptions, we should probably use the same param for all other tasks

File: src/Raven.Client/Documents/Operations/OngoingTasks/GetOngoingTaskInfoOperation.cs [hunk 2]
```diff
@@ -34,9 +44,15 @@ public GetOngoingTaskInfoCommand(long taskId, OngoingTaskType type)
...
+            public GetOngoingTaskInfoCommand(string taskName, OngoingTaskType type)
+            {
+                _taskName = taskName;
+                _type = type;
+            }
+
             public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
             {
-                url = $"{node.Url}/databases/{node.Database}/task?key={_taskId}&type={_type}";
+                url = _taskName != null ? $"{node.Url}/databases/{node.Database}/task?name={_taskName}&type={_type}" : $"{node.Url}/databases/{node.Database}/task?key={_taskId}&type={_type}";
```
Comment (ayende) [inline src/Raven.Client/Documents/Operations/OngoingTasks/GetOngoingTaskInfoOperation.cs]: Can you break this on multiple lines?

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 3]
```diff
@@ -829,8 +829,15 @@ public Task GetOngoingTaskInfo()
...
             if (ResourceNameValidator.IsValidResourceName(Database.Name, ServerStore.Configuration.Core.DataDirectory.FullPath, out string errorMessage) == false)
                 throw new BadRequestException(errorMessage);
+            long key = 0;
+            var taskId = GetLongQueryString("key", false);
+            if (taskId != null)
+                key = taskId.Value;
+            var name = GetStringQueryString("name", false);
+
+            if ((taskId == null) && (name == null))
+                throw new ArgumentException($"Query string key or name is mandatory, but wasn't specified.");
```
Comment (ayende) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs]:
```suggestion
                throw new ArgumentException($"You must specify a query string argument of either 'key' or 'name' , but none was specified.");
```

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Operations/OngoingTasks/GetOngoingTaskInfoOperation.cs [hunk 4]
```diff
@@ -18,13 +19,22 @@ public GetOngoingTaskInfoOperation(long taskId, OngoingTaskType type)
             _type = type;
         }
 
+        public GetOngoingTaskInfoOperation(string taskName, OngoingTaskType type)
+        {
+            _taskName = taskName;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/OngoingTasks/GetOngoingTaskInfoOperation.cs:24]: assert that string is not null of whitespace

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 5]
```diff
@@ -851,7 +858,10 @@ public Task GetOngoingTaskInfo()
                     {
                         case OngoingTaskType.Replication:
 
-                            var watcher = record.ExternalReplications.Find(x => x.TaskId == key);
+                            var watcher = name != null ?
+                                record.ExternalReplications.Find(x => x.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase))
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs]: `StringComparison.OridinalIgnoreCase`

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 6]
```diff
@@ -865,7 +875,10 @@ public Task GetOngoingTaskInfo()
 
                         case OngoingTaskType.Backup:
 
-                            var backupConfiguration = record.PeriodicBackups?.Find(x => x.TaskId == key);
+                            var backupConfiguration = name != null ?
+                                record.PeriodicBackups.Find(x => x.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase))
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs]: `StringComparison.OridinalIgnoreCase`

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 7]
```diff
@@ -879,7 +892,10 @@ public Task GetOngoingTaskInfo()
 
                         case OngoingTaskType.SqlEtl:
 
-                            var sqlEtl = record.SqlEtls?.Find(x => x.TaskId == key);
+                            var sqlEtl = name != null ?
+                                record.SqlEtls.Find(x => x.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase))
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs]: `StringComparison.OridinalIgnoreCase`

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 8]
```diff
@@ -904,7 +920,10 @@ public Task GetOngoingTaskInfo()
 
                         case OngoingTaskType.RavenEtl:
 
-                            var ravenEtl = record.RavenEtls?.Find(x => x.TaskId == key);
+                            var ravenEtl = name != null ?
+                                record.RavenEtls.Find(x => x.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase))
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs]: `StringComparison.OridinalIgnoreCase`

**Review by ppekrol — APPROVED** [review]

**Pull request merged**